integration/index_test.go

```diff
@@ -0,0 +1,179 @@
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"encoding/json"
+	"strings"
+	"testing"
+
+	"github.com/awslabs/soci-snapshotter/soci"
+	"github.com/awslabs/soci-snapshotter/util/dockershell"
+	"github.com/containerd/containerd/platforms"
+)
+
+type testImageIndex struct {
+	imgName         string
+	platform        string
+	imgInfo         imageInfo
+	sociIndexDigest string
+}
+
+func prepareSociIndices(t *testing.T, sh *dockershell.Shell) []testImageIndex {
+	testImages := []testImageIndex{
+		{
+			imgName:  "ubuntu:latest",
+			platform: "linux/arm64",
+		},
+		{
+			imgName:  "alpine:latest",
+			platform: "linux/amd64",
+		},
+		{
+			imgName:  "nginx:latest",
+			platform: "linux/arm64",
+		},
+		{
+			imgName:  "drupal:latest",
+			platform: "linux/amd64",
+		},
+	}
+
+	for i, img := range testImages {
+		platform := platforms.DefaultSpec()
+		if img.platform != "" {
+			var err error
+			platform, err = platforms.Parse(img.platform)
+			if err != nil {
+				t.Fatalf("could not parse platform: %v", err)
+			}
+		}
+		img.imgInfo = dockerhub(img.imgName, withPlatform(platform))
+		img.sociIndexDigest = optimizeImage(sh, img.imgInfo)
+		testImages[i] = img
+	}
+
+	return testImages
+}
+
+func TestSociIndexInfo(t *testing.T) {
+	t.Parallel()
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+	rebootContainerd(t, sh, "", "")
+
+	testImages := prepareSociIndices(t, sh)
+
+	for _, img := range testImages {
+		t.Run(img.imgName, func(t *testing.T) {
+			var sociIndex soci.Index
+			rawJSON := sh.O("soci", "index", "info", img.sociIndexDigest)
+			if err := json.Unmarshal(rawJSON, &sociIndex); err != nil {
+				t.Fatalf("invalid soci index from digest %s: %v", img.sociIndexDigest, rawJSON)
+			}
+
+			m, err := getManifestDigest(sh, img.imgInfo.ref, img.imgInfo.platform)
+			if err != nil {
+				t.Fatalf("failed to get manifest digest: %v", err)
+			}
+
+			validateSociIndex(t, sh, sociIndex, m, nil)
+		})
+	}
+}
+
+func TestSociIndexList(t *testing.T) {
+	t.Parallel()
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+	rebootContainerd(t, sh, "", "")
+
+	testImages := prepareSociIndices(t, sh)
+
+	existHandlerFull := func(output string, img testImageIndex) bool {
+		// full output should have both img ref and soci index digest
+		return strings.Contains(output, img.imgInfo.ref) && strings.Contains(output, img.sociIndexDigest)
+	}
+
+	existHandlerQuiet := func(output string, img testImageIndex) bool {
+		// a given soci index should match exactly one line in the quiet output
+		// for the first index, it should have prefix of digest+\n
+		// for the rest, it should have `\n` before and after its digest
+		return strings.HasPrefix(output, img.sociIndexDigest+"\n") || strings.Contains(output, "\n"+img.sociIndexDigest+"\n")
```

Comment:
All this string manipulation seems flaky to me. What if we decide to change the output? I think it make sense to have `--json` output. Doesn't have to be part of this PR, but something to think about.